### ISSUE TITLE
IndexError: list index out of range

### ISSUE THREAD

**Issue by cyring** [sphinx-doc/sphinx#10495]
### Describe the bug

make[1]: *** [Documentation/Makefile:96: htmldocs] Error 2
make: *** [Makefile:1678: htmldocs] Error 2
==> ERROR: A failure occurred in build().
    Aborting...

### How to Reproduce

[Arch Build System](https://wiki.archlinux.org/title/Arch_Build_System)
```
$ cd linux
$ asp checkout linux
$ asp update && git pull
$ cd repos/core-x86_64/
$ makepkg --noconfirm -cCsfir
```

### Expected behavior

1. `linux-5.18.arch1-1-x86_64.pkg.tar.zst`
2. `linux-docs-5.18.arch1-1-x86_64.pkg.tar.zst`
3. `linux-headers-5.18.arch1-1-x86_64.pkg.tar.zst`

### Your project

archlinux

### Screenshots

```
tail -n 50 /tmp/sphinx-err-irzy92f8.log
```
----
```
#   writing output... [  5%] admin-guide/device-mapper/dm-integrity .. admin-guide/gpio/gpio-aggregator
#   writing output... [  6%] admin-guide/gpio/gpio-mockup .. admin-guide/laptops/lg-laptop
#   writing output... [  7%] admin-guide/laptops/sony-laptop .. admin-guide/media/dvb-usb-af9035-cardlist
#   writing output... [  8%] admin-guide/media/dvb-usb-anysee-cardlist .. admin-guide/media/dvb-usb-zd1301-cardlist
# Loaded extensions:
#   sphinx.ext.mathjax (5.0.0) from /usr/lib/python3.10/site-packages/sphinx/ext/mathjax.py
#   sphinxcontrib.applehelp (1.0.2) from /usr/lib/python3.10/site-packages/sphinxcontrib/applehelp/__init__.py
#   sphinxcontrib.devhelp (1.0.2) from /usr/lib/python3.10/site-packages/sphinxcontrib/devhelp/__init__.py
#   sphinxcontrib.htmlhelp (2.0.0) from /usr/lib/python3.10/site-packages/sphinxcontrib/htmlhelp/__init__.py
#   sphinxcontrib.serializinghtml (1.1.5) from /usr/lib/python3.10/site-packages/sphinxcontrib/serializinghtml/__init__.py
#   sphinxcontrib.qthelp (1.0.3) from /usr/lib/python3.10/site-packages/sphinxcontrib/qthelp/__init__.py
#   alabaster (0.7.12) from /usr/lib/python3.10/site-packages/alabaster/__init__.py
#   kerneldoc (1.0) from~/pkg/PKGBUILD/linux/repos/core-x86_64/src/archlinux-linux/Documentation/sphinx/kerneldoc.py
#   rstFlatTable (1.0) from ~/pkg/PKGBUILD/linux/repos/core-x86_64/src/archlinux-linux/Documentation/sphinx/rstFlatTable.py
#   kernel_include (1.0) from ~/pkg/PKGBUILD/linux/repos/core-x86_64/src/archlinux-linux/Documentation/sphinx/kernel_include.py
#   kfigure (1.0.0) from ~/pkg/PKGBUILD/linux/repos/core-x86_64/src/archlinux-linux/Documentation/sphinx/kfigure.py
#   sphinx.ext.ifconfig (5.0.0) from /usr/lib/python3.10/site-packages/sphinx/ext/ifconfig.py
#   automarkup (unknown version) from ~/pkg/PKGBUILD/linux/repos/core-x86_64/src/archlinux-linux/Documentation/sphinx/automarkup.py
#   maintainers_include (1.0) from ~/pkg/PKGBUILD/linux/repos/core-x86_64/src/archlinux-linux/Documentation/sphinx/maintainers_include.py
#   kernel_abi (1.0) from ~/pkg/PKGBUILD/linux/repos/core-x86_64/src/archlinux-linux/Documentation/sphinx/kernel_abi.py
#   kernel_feat (1.0) from ~/pkg/PKGBUILD/linux/repos/core-x86_64/src/archlinux-linux/Documentation/sphinx/kernel_feat.py
#   sphinx.ext.imgmath (5.0.0) from /usr/lib/python3.10/site-packages/sphinx/ext/imgmath.py
Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/sphinx/cmd/build.py", line 276, in build_main
    app.build(args.force_all, filenames)
  File "/usr/lib/python3.10/site-packages/sphinx/application.py", line 329, in build
    self.builder.build_update()
  File "/usr/lib/python3.10/site-packages/sphinx/builders/__init__.py", line 288, in build_update
    self.build(to_build,
  File "/usr/lib/python3.10/site-packages/sphinx/builders/__init__.py", line 352, in build
    self.write(docnames, list(updated_docnames), method)
  File "/usr/lib/python3.10/site-packages/sphinx/builders/__init__.py", line 541, in write
    self._write_parallel(sorted(docnames),
  File "/usr/lib/python3.10/site-packages/sphinx/builders/__init__.py", line 585, in _write_parallel
    doctree = self.env.get_and_resolve_doctree(docname, self)
  File "/usr/lib/python3.10/site-packages/sphinx/environment/__init__.py", line 530, in get_and_resolve_doctree
    self.apply_post_transforms(doctree, docname)
  File "/usr/lib/python3.10/site-packages/sphinx/environment/__init__.py", line 576, in apply_post_transforms
    transformer.apply_transforms()
  File "/usr/lib/python3.10/site-packages/sphinx/transforms/__init__.py", line 80, in apply_transforms
    super().apply_transforms()
  File "/usr/lib/python3.10/site-packages/docutils/transforms/__init__.py", line 171, in apply_transforms
    transform.apply(**kwargs)
  File "/usr/lib/python3.10/site-packages/sphinx/transforms/post_transforms/__init__.py", line 35, in apply
    self.run(**kwargs)
  File "/usr/lib/python3.10/site-packages/sphinx/builders/html/transforms.py", line 44, in run
    parts = self.pattern.split(node[-1].astext())
  File "/usr/lib/python3.10/site-packages/docutils/nodes.py", line 654, in __getitem__
    return self.children[key]
IndexError: list index out of range
```

### OS

Linux RYZEN 5.18.0-arch1-1 x86_64

### Python version

`Python 3.10.4`

### Sphinx version

`sphinx-build 5.0.0`

### Sphinx extensions

`sphinx-autogen 5.0.0` , `sphinx-apidoc 5.0.0` , `sphinx-quickstart 5.0.0`

### Extra tools

`gcc (GCC) 12.1.0` , `GNU Make 4.3` , `ldd (GNU libc) 2.35`, `git version 2.36.1`, `GNU bash, version 5.1.16(1)`

**Comment by aaime** [comment]
Getting the same in a Github Action, started happening yesterday, as the build switched to Sphinx 5:

```
sphinx:
     [echo] Running sphinx-build -D release=28-SNAPSHOT -q -W --keep-going -j auto -b html -d "/home/runner/work/geotools/geotools/docs/target/developer/doctrees" . "/home/runner/work/geotools/geotools/docs/target/developer/html"
     [exec] 
     [exec] Exception occurred:
     [exec]   File "/opt/hostedtoolcache/Python/3.10.4/x64/lib/python3.10/site-packages/docutils/nodes.py", line 654, in __getitem__
     [exec]     return self.children[key]
     [exec] IndexError: list index out of range
     [exec] The full traceback has been saved in /tmp/sphinx-err-f7bvx8hf.log, if you want to report the issue to the developers.
     [exec] Please also report this if it was a user error, so that a better error message can be provided next time.
     [exec] A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
```

**Comment by AA-Turner** [comment]
Hi @cyring  / @aaime -- is your project the Linux kernel? I can't really reproduce this on Windows.

Can you try deleting as much of the docs as possible before it stops failing for a minimal reproducer? It's likely we'll do a [`5.0.1` release](https://github.com/sphinx-doc/sphinx/milestone/127) as there are a few other bugs.

A

**Comment by aaime** [comment]
@AA-Turner in my case it's the [GeoTools documentation](https://github.com/geotools/geotools/tree/main/docs).
Unfortunately, I cannot reproduce it on my local machine, it only happens on the Github actions... I've tried for a while but don't know enough about python and its dependency management (or at least, that's why I think I'm not able to).

**Comment by cyring** [comment]
> Hi @cyring / @aaime -- is your project the Linux kernel? I can't really reproduce this on Windows.
> 
> Can you try deleting as much of the docs as possible before it stops failing for a minimal reproducer? It's likely we'll do a [`5.0.1` release](https://github.com/sphinx-doc/sphinx/milestone/127) as there are a few other bugs.
> 
> A

Hello,

The purpose is to build the latest mainstream Linux kernel. 
The working operating system is Linux the from the Arch Linux distribution which provides some scripts, named ASP, to build and install the next kernel release: a rolling release.
Among those scripts, is run the kernel html doc which is Sphinx based. That part is failing.

Fyi, I have been doing such kernel build once a week for months/years and this is the first time such issue is encountered.
Be also aware, as rolling release, Arch brings the latest stable software packages and thus Sphinx dependencies, like Python.

Hope it helps, feel free to ask other details.

**Comment by AA-Turner** [comment]
@cyring Sphinx should save a full log on failure, please could you upload it here?

A

**Comment by AA-Turner** [comment]
I found a minimal reproducer from @aaime's GeoTools documentation (after working out how to read an XML build script!!):

```python
import shutil
from pathlib import Path

from sphinx.cmd.make_mode import run_make_mode


def write(filename, text): Path(filename).write_text(text, encoding="utf-8")

write("conf.py", '''\
''')

write("index.rst", '''\
:kbd:`blah - blah`
''')

shutil.rmtree("_build", ignore_errors=True)
run_make_mode(["html", ".", "_build", "-T", "-W"])
```

run as `reproducer_10495.py`. The `blah - blah` is important -- there must be a hypen and at least one space. @cyring can you check if a similar construct appears in the Linux docs? (`` :kbd:`[a-z0-9 ]*?( \-| \-| \- )[a-z0-9 ]*?` ``)

A

**Comment by cyring** [comment]
> @cyring Sphinx should save a full log on failure, please could you upload it here?
> 
> A

Please find the log below:
```
# Sphinx version: 5.0.0
# Python version: 3.10.4 (CPython)
# Docutils version: 0.18.1 release
# Jinja2 version: 3.1.2
# Last messages:
#   preparing documents...
#   done
#   writing output... [  1%] PCI/boot-interrupts .. RCU/rcu
#   writing output... [  2%] RCU/rcu_dereference .. admin-guide/acpi/index
#   writing output... [  3%] admin-guide/acpi/initrd_table_override .. admin-guide/cgroup-v1/devices
#   writing output... [  4%] admin-guide/cgroup-v1/freezer-subsystem .. admin-guide/device-mapper/dm-init
#   writing output... [  5%] admin-guide/device-mapper/dm-integrity .. admin-guide/gpio/gpio-aggregator
#   writing output... [  6%] admin-guide/gpio/gpio-mockup .. admin-guide/laptops/lg-laptop
#   writing output... [  7%] admin-guide/laptops/sony-laptop .. admin-guide/media/dvb-usb-af9035-cardlist
#   writing output... [  8%] admin-guide/media/dvb-usb-anysee-cardlist .. admin-guide/media/dvb-usb-zd1301-cardlist
# Loaded extensions:
#   sphinx.ext.mathjax (5.0.0) from /usr/lib/python3.10/site-packages/sphinx/ext/mathjax.py
#   sphinxcontrib.applehelp (1.0.2) from /usr/lib/python3.10/site-packages/sphinxcontrib/applehelp/__init__.py
#   sphinxcontrib.devhelp (1.0.2) from /usr/lib/python3.10/site-packages/sphinxcontrib/devhelp/__init__.py
#   sphinxcontrib.htmlhelp (2.0.0) from /usr/lib/python3.10/site-packages/sphinxcontrib/htmlhelp/__init__.py
#   sphinxcontrib.serializinghtml (1.1.5) from /usr/lib/python3.10/site-packages/sphinxcontrib/serializinghtml/__init__.py
#   sphinxcontrib.qthelp (1.0.3) from /usr/lib/python3.10/site-packages/sphinxcontrib/qthelp/__init__.py
#   alabaster (0.7.12) from /usr/lib/python3.10/site-packages/alabaster/__init__.py
#   kerneldoc (1.0) from ~/pkg/PKGBUILD/linux/repos/core-x86_64/src/archlinux-linux/Documentation/sphinx/kerneldoc.py
#   rstFlatTable (1.0) from ~/pkg/PKGBUILD/linux/repos/core-x86_64/src/archlinux-linux/Documentation/sphinx/rstFlatTable.py
#   kernel_include (1.0) from ~/pkg/PKGBUILD/linux/repos/core-x86_64/src/archlinux-linux/Documentation/sphinx/kernel_include.py
#   kfigure (1.0.0) from ~/pkg/PKGBUILD/linux/repos/core-x86_64/src/archlinux-linux/Documentation/sphinx/kfigure.py
#   sphinx.ext.ifconfig (5.0.0) from /usr/lib/python3.10/site-packages/sphinx/ext/ifconfig.py
#   automarkup (unknown version) from ~/pkg/PKGBUILD/linux/repos/core-x86_64/src/archlinux-linux/Documentation/sphinx/automarkup.py
#   maintainers_include (1.0) from ~/pkg/PKGBUILD/linux/repos/core-x86_64/src/archlinux-linux/Documentation/sphinx/maintainers_include.py
#   kernel_abi (1.0) from ~/pkg/PKGBUILD/linux/repos/core-x86_64/src/archlinux-linux/Documentation/sphinx/kernel_abi.py
#   kernel_feat (1.0) from ~/pkg/PKGBUILD/linux/repos/core-x86_64/src/archlinux-linux/Documentation/sphinx/kernel_feat.py
#   sphinx.ext.imgmath (5.0.0) from /usr/lib/python3.10/site-packages/sphinx/ext/imgmath.py
Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/sphinx/cmd/build.py", line 276, in build_main
    app.build(args.force_all, filenames)
  File "/usr/lib/python3.10/site-packages/sphinx/application.py", line 329, in build
    self.builder.build_update()
  File "/usr/lib/python3.10/site-packages/sphinx/builders/__init__.py", line 288, in build_update
    self.build(to_build,
  File "/usr/lib/python3.10/site-packages/sphinx/builders/__init__.py", line 352, in build
    self.write(docnames, list(updated_docnames), method)
  File "/usr/lib/python3.10/site-packages/sphinx/builders/__init__.py", line 541, in write
    self._write_parallel(sorted(docnames),
  File "/usr/lib/python3.10/site-packages/sphinx/builders/__init__.py", line 585, in _write_parallel
    doctree = self.env.get_and_resolve_doctree(docname, self)
  File "/usr/lib/python3.10/site-packages/sphinx/environment/__init__.py", line 530, in get_and_resolve_doctree
    self.apply_post_transforms(doctree, docname)
  File "/usr/lib/python3.10/site-packages/sphinx/environment/__init__.py", line 576, in apply_post_transforms
    transformer.apply_transforms()
  File "/usr/lib/python3.10/site-packages/sphinx/transforms/__init__.py", line 80, in apply_transforms
    super().apply_transforms()
  File "/usr/lib/python3.10/site-packages/docutils/transforms/__init__.py", line 171, in apply_transforms
    transform.apply(**kwargs)
  File "/usr/lib/python3.10/site-packages/sphinx/transforms/post_transforms/__init__.py", line 35, in apply
    self.run(**kwargs)
  File "/usr/lib/python3.10/site-packages/sphinx/builders/html/transforms.py", line 44, in run
    parts = self.pattern.split(node[-1].astext())
  File "/usr/lib/python3.10/site-packages/docutils/nodes.py", line 654, in __getitem__
    return self.children[key]
IndexError: list index out of range
```

**Comment by AA-Turner** [comment]
Thanks @cyring, it seems to be the same `:kbd:` failure that should be fixed in #10504 -- if you'd be able to test the fix I would really appreciate it.

A

**Comment by cyring** [comment]
> Thanks @cyring, it seems to be the same `:kbd:` failure that should be fixed in #10504 -- if you'd be able to test the fix I would really appreciate it.
> 
> A

That fix made it successfully.

```sh
pacman -Ql python-sphinx | grep transforms.py
```
----
```
python-sphinx /usr/lib/python3.10/site-packages/sphinx/builders/html/transforms.py
python-sphinx /usr/lib/python3.10/site-packages/sphinx/builders/latex/transforms.py
```
----
```sh
nano -l /usr/lib/python3.10/site-packages/sphinx/builders/html/transforms.py
```
![2022-06-01-022823_706x665_scrot](https://user-images.githubusercontent.com/11563789/171304773-705ebeab-9878-4e75-843e-e52db8bb9c70.png)
----
[Here](https://archlinux.org/packages/community/any/python-sphinx/) the ArchLinux Maintainers of `python-sphinx`

**Comment by tk0miya** [comment]
Fixed by https://github.com/sphinx-doc/sphinx/pull/10504. The fixed version will be released soon. Please wait for a moment.
Thank you for reporting
